server/aesdsocket.c

```diff
@@ -0,0 +1,352 @@
+    }
+
+    //Remove the output file
+    if (remove(OUTPUT_FILE) == 0) {
+        printf("File '%s' deleted successfully.\n", OUTPUT_FILE);
+    }
+
+
+
+    //First get the available addresses on the host at port 9000
+    memset(&hints, 0, sizeof hints);
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE; // use my IP
+
+    int rc = getaddrinfo(NULL, PORT, &hints, &address_results);
+
+    if (rc != 0) {
+        //Handle Error
+        fprintf(stderr, "Failed to get address info Error: %s\n", gai_strerror(rc));
+        return ERROR_RESULT;
+    }
+    //Iterate through and bind
+    for (nodes = address_results; nodes != NULL; nodes = nodes->ai_next) {
+        if ((socket_fd = socket(nodes->ai_family, nodes->ai_socktype,
+                                nodes->ai_protocol)) == -1) {
+            perror("server: socket");
+            continue;
+        }
+
+        if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &yes,
+                       sizeof(int)) == -1) {
+            perror("setsockopt");
+            exit(1);
+        }
+
+        if (bind(socket_fd, nodes->ai_addr, nodes->ai_addrlen) == -1) {
+            close(socket_fd);
+            perror("server: bind");
+            continue;
+        }
+
+        break;
+    }
+
+    freeaddrinfo(address_results);
+
+    //Make sure bind was successful
+    if (nodes == NULL) {
+        fprintf(stderr, "server: failed to bind\n");
+        return ERROR_RESULT;
+    }
+	//since bind was successful now lets start as a daemon
+	if(daemon_mode){
+		daemonize();
+	}
+
+    //Start listening using socket
+    if (listen(socket_fd, BACKLOG) == -1) {
+        perror("failed to listen to connection");
+        return ERROR_RESULT;
+    }
+
+//    signal_action.sa_handler = signal_handler;
+//    sigemptyset(&signal_action.sa_mask);
+//    signal_action.sa_flags = SA_RESTART;
+//    if (sigaction(SIGCHLD, &signal_action, NULL) == -1) {
+//        perror("sigaction");
+//        exit(1);
+//    }
+    printf("Currently listening for connections!\n");
+
+    while (1) {  // main accept() loop
+        sin_size = sizeof their_addr;
+        client_socket = accept(socket_fd, (struct sockaddr *) &their_addr, &sin_size);
+        if (client_socket == -1) {
+            perror("accept");
+            continue;
+        }
+
+        inet_ntop(their_addr.ss_family,
+                  get_in_addr((struct sockaddr *) &their_addr),
+                  s, sizeof s);
+        printf("server: got connection from %s \r\n", s);
+        syslog(LOG_INFO, "Accepted connection from %s \r\n", s);
+
+
+        
+        bool end_of_packet = false;
+        uint32_t packet_buffer_pos = 0;
+
+
+        char buffer[BUFFER_SIZE*20];
+
+        ssize_t bytes_received;
+
+        //Allocate for the packet buffer
+        packet_buffer = (char *) malloc(BUFFER_SIZE*20);
+
+        while (1) {
+            bytes_received = recv(client_socket, buffer, BUFFER_SIZE, 0);
```

Comment:
@Silex93 what happens if the total size of the packet before `\n` is > `BUFFER_SIZE`?